test/sql/hnsw_index_from_file.sql

```diff
@@ -0,0 +1,30 @@
+------------------------------------------------------------------------------
+-- Test HNSW index creation from file
+------------------------------------------------------------------------------
+
+-- Index files were created with ldb-create-index program which source is under https://github.com/lanterndata/lanterndb_extras/
+-- We have exported index files for sift1k dataset for cosine and l2sq distances
+-- With the following params m=16 ef=64 ef_construction=128 dims=128
+-- Validate that index creation works with a small number of vectors
+\ir utils/sift1k_array.sql
+
+-- Validate that creating an index from file works
+CREATE INDEX hnsw_l2_index ON sift_base1k USING hnsw (v dist_l2sq_ops) WITH (dims=128, M=16, ef=64, ef_construction=128, _experimental_index_path='/tmp/lanterndb/files/index-sift1k-l2.usearch');
+SELECT * FROM ldb_get_indexes('sift_base1k');
+
+SET enable_seqscan = false;
+
+SELECT v AS v777 FROM sift_base1k WHERE id = 777 \gset
+EXPLAIN (COSTS FALSE) SELECT ROUND(l2sq_dist(v, :'v777')::numeric, 2) FROM sift_base1k order by v <-> :'v777' LIMIT 10;
+SELECT ROUND(l2sq_dist(v, :'v777')::numeric, 2) FROM sift_base1k order by v <-> :'v777' LIMIT 10;
+
+DROP INDEX hnsw_l2_index;
+
+-- Validate that creating an index from file works with cosine distance function
+CREATE INDEX hnsw_cos_index ON sift_base1k USING hnsw (v dist_cos_ops) WITH (dims=128, M=16, ef=64, ef_construction=128, _experimental_index_path='/tmp/lanterndb/files/index-sift1k-cos.usearch');
+SELECT * FROM ldb_get_indexes('sift_base1k');
+
+SELECT v AS v777 FROM sift_base1k WHERE id = 777 \gset
+EXPLAIN (COSTS FALSE) SELECT ROUND(cos_dist(v, :'v777')::numeric, 2) FROM sift_base1k order by v <-> :'v777' LIMIT 10;
+SELECT ROUND(cos_dist(v, :'v777')::numeric, 2) FROM sift_base1k order by v <-> :'v777' LIMIT 10;
+
```

Comment:
Consider these cases:

case 1:

1) Index is created externally.
2) More vectors are added to the table
3) CREATE INDEX is run on the table with the external file

case 2:
1) Index is created externally
2) Vectors are deleted from the table
3) CREATE iNDEX is run on the table with external file

case 3:
1) Index is created externally
2) Vectors updated
3) CREATE iNDEX is run on the table with external file


I assume currently case1 results in certain vectors not being in the index and never being returned by search queries. If so, let's add the test to the `todos` regression test so we have it tracked. If the case is handled, let's have a test for it.

I think case2 is currently handled fine. Our index will have extra dead entries but that is fine. We can deal with it when we implement proper FULL VACUUM. Can you check this and add a test case for it?

case 3: I think we should explicitly ban this in external index API definition in our docs. That said, I think this works fine in most cases since I think UPDATEs result in a new tuple being created with a fresh ctid. (Things probably break after a VACUUM that is why we should ban this in docs)

Reply:
Caveat: I understand that case 3 should be banned for now until deletes and inserts are working.

I would think that case 3 works as long as case 1 and 2 work, since update = delete and insert as far as Postgres is considered. The index would have a pointer to a dead tuple, which would be vacuumed. Whatever means we use to add newly inserted vectors in case 1, I would expect to also apply to case 3.